jormungandr/src/fragment/pool.rs

```diff
@@ -68,6 +76,21 @@ impl Pools {
             .filter(|(_, exists_in_logs)| !exists_in_logs)
             .map(|(fragment, _)| fragment);
 
+        if let Some(mut persistent_log) = self.persistent_log.as_mut() {
```

Comment:
I actually thought we were going to put this at the API level to bypass any check that is done on the node

Reply:
We discussed that with @mzabaluev and decided to run it only for fragments that made it to the log. With #3232 we will have proper errors for what did not made its way to logs. And apart from pool overflow these errors are unlikely to happen. The overflow is also pretty unlikely in the production setup from what I have seen.

Reply:
This is also filtering for already existing fragments, but it may be intended and beneficial to reduce the amount of stuff we store

Reply:
For ballot fragments in particular, we can't tell if a repeated fragment is a replay attempt, or is it the combination of a user changing their mind on a ballot at least 2 times, and the client wrongly reusing the spending counter from the first fragment. I consider the likelihood of the latter low enough to matter, so we can ignore repeated fragments.